src/services/communications/Packets.py

```diff
@@ -0,0 +1,90 @@
+                self._dict_packets[id].append(data_type)
+
+    
+    def validate_packet(self,packet_id,values):
+        #validate id
+        if packet_id not in self._dict_packets:
+            raise KeyError(f"Packet ID {packet_id} not found in the map")
+        
+        expected_sizes = self._dict_packets[packet_id]    
+        if len(values) != len(expected_sizes):
+            raise ValueError(f"Expected {len(expected_sizes)} but received {len(values)}")
+        #validate types
+        for value,data_type in zip(values,self._dict_packets[packet_id]):
+            match data_type:
+                case "uint8" | "enum":
+                    if not (0 <= value <= 255) or not isinstance(value,int):
+                        raise ValueError(f"Value {value} out of range for uint8 or not the data_type expected")
+                case "int8":
+                    if not (-128 <= value <= 127) or not isinstance(value,int):
+                        raise ValueError(f"Value {value} out of range for int8 or not the data_type expected")
+                case "uint16":
+                    if not (0 <= value <= 0xFFFF) or not isinstance(value,int):
+                        raise ValueError(f"Value {value} out of range for uint16 or not the data_type expected")
+                case "int16":
+                    if not (-0x8000 <= value <= 0x7FFF) or not isinstance(value,int):
+                        raise ValueError(f"Value {value} out of range for int16 or not the data_type expected")
+                case "uint32":
+                    if not (0 <= value <= 0xFFFFFFFF) or not isinstance(value,int):
+                        raise ValueError(f"Value {value} out of range for uint32 or not the data_type expected")
+                case "int32":
+                    if not (-0x80000000 <= value <= 0x7FFFFFFF) or not isinstance(value,int):
+                        raise ValueError(f"Value {value} out of range for int32 or not the data_type expected")
+                case "float32":
+                    if not (isinstance(value, float) or isinstance(value, int)):
+                        raise ValueError(f"Value {value} is not a valid float32")
+                case "uint64":
+                    if not (0 <= value <= 0xFFFFFFFFFFFFFFFF) or not isinstance(value,int):
+                        raise ValueError(f"Value {value} out of range for uint64 or not the data_type expected")
+                case "int64":
+                    if not (-0x8000000000000000 <= value <= 0x7FFFFFFFFFFFFFFF) or not isinstance(value,int):
+                        raise ValueError(f"Value {value} out of range for int64 or not the data_type expected")
+                case "float64":
+                    if not (isinstance(value, float) or isinstance(value, int)):
+                        raise ValueError(f"Value {value} is not a valid float64")
+                case _:
+                    raise ValueError(f"Unsupported data type: {data_type}")
```

Comment:
instead of having one big function to check this, have one function for each type and call the appropriate function as you are serializing the packet, that way you don't have to iterate twice over all the data types of the packet.